react/features/transcribing/middleware.ts

```diff
@@ -29,62 +29,82 @@ const TRANSCRIBER_DISPLAY_NAME = 'Transcriber';
         potentialTranscriberJIDs
-    } = store.getState()['features/transcribing'];
+    } = getState()['features/transcribing'];
 
     switch (action.type) {
+    case _TRANSCRIBER_JOINED: {
+        notifyTranscribingStatusChanged(true);
+        dispatch(showStartedTranscribingNotification());
+
+        const state = getState();
+        const { transcription } = state['features/base/config'];
+        const { _requestingSubtitles } = state['features/subtitles'];
+
+        if (!_requestingSubtitles && !transcription?.disableStartForAll) {
+            dispatch(toggleRequestingSubtitles());
+        }
+        break;
+    }
     case _TRANSCRIBER_LEFT: {
-        store.dispatch(showStoppedTranscribingNotification());
-        const state = store.getState();
+        notifyTranscribingStatusChanged(false);
+        dispatch(showStoppedTranscribingNotification());
+
+        const state = getState();
         const { transcription } = state['features/base/config'];
         const { _requestingSubtitles } = state['features/subtitles'];
 
         if (_requestingSubtitles && !transcription?.disableStartForAll) {
-            store.dispatch(toggleRequestingSubtitles());
+            dispatch(toggleRequestingSubtitles());
         }
         break;
     }
     case HIDDEN_PARTICIPANT_JOINED:
-        if (action.displayName
-                && action.displayName === TRANSCRIBER_DISPLAY_NAME) {
-            store.dispatch(transcriberJoined(action.id));
+        if (action.displayName === TRANSCRIBER_DISPLAY_NAME) {
+            dispatch(transcriberJoined(action.id));
         } else {
-            store.dispatch(potentialTranscriberJoined(action.id));
+            dispatch(potentialTranscriberJoined(action.id));
         }
 
         break;
     case HIDDEN_PARTICIPANT_LEFT:
         if (action.id === transcriberJID) {
-            store.dispatch(transcriberLeft(action.id));
+            dispatch(transcriberLeft(action.id));
         }
         break;
     case PARTICIPANT_UPDATED: {
         const { participant } = action;
 
-        if (potentialTranscriberJIDs.includes(participant.id)
-            && participant.name === TRANSCRIBER_DISPLAY_NAME) {
-            batch(() => {
-                store.dispatch(transcriberJoined(participant.id));
-                store.dispatch(hidePendingTranscribingNotification());
-            });
+        if (potentialTranscriberJIDs.includes(participant.id) && participant.name === TRANSCRIBER_DISPLAY_NAME) {
+            dispatch(transcriberJoined(participant.id));
         }
 
         break;
     }
-    case _TRANSCRIBER_JOINED: {
-        const state = store.getState();
-        const { transcription } = state['features/base/config'];
-        const { _requestingSubtitles } = state['features/subtitles'];
-
-        if (!_requestingSubtitles && !transcription?.disableStartForAll) {
-            store.dispatch(toggleRequestingSubtitles());
+    case SET_REQUESTING_SUBTITLES:
+        if (!isTranscribing) {
```

Comment:
shouldn't this also depend on `action.enabled` being `true`?

Reply:
Good catch!